### DISCUSSION TITLE
v1.0.1.0

### DISCUSSION THREAD

**Discussion by PeterSuh-Q3** [PeterSuh-Q3/tinycore-redpill#19] [Q&A] [answered]
1.0.1.0 Upgrade from Tinycore version 12.0 (kernel 5.10.3) to 14.0 (kernel 6.1.2) 
    to improve compatibility with the latest devices.

![스크린샷 2023-12-18 오후 12 57 10](https://github.com/PeterSuh-Q3/tinycore-redpill/assets/85427533/f9d41591-a81b-4612-825c-6283e0f41f80)

<hr /><em>This discussion was created from the release <a href='https://github.com/PeterSuh-Q3/tinycore-redpill/releases/tag/v1.0.1.0'>v1.0.1.0</a>.</em>

**Comment by TheNighthawk99** [comment]
Hello,
I have been following this project for many months now and I wanna thank you for your great effort.
I have noted that the size of versions 1.0.1 and 1.0.0.5 are almost 4x the size of versions <1, which are about 200 MiB.
Why this big difference in terms of file size?

Thank you.
Regards.

**Reply by PeterSuh-Q3** [reply]
First of all, the basic size of the img file has increased from 1G to 1.5G.
And, the default content in the img file is gradually increasing.

The most important thing is that when creating this img file from a USB stick, "Win32 Disk Imager 1.0" is used, and the compression function of this tool is utilized.
When using compression, if there are any fragments in the partition contents of the USB stick, it seems to reduce compression efficiency.

I'm not sure how to collect the fragments of the USB stick partition.
If this becomes possible, compression efficiency will likely be improved.

**Reply by TheNighthawk99** [reply]
Many thanks for your reply.

**Reply by PeterSuh-Q3** [reply]
You have succeeded in making the gz compressed files lightweight.
It is available starting from the next version, v1.0.2.4.

https://github.com/PeterSuh-Q3/tinycore-redpill/releases/tag/v1.0.2.4

**Comment by Daniel042001** [comment]
![1192dae0930933122a1a0bd9a8560f0b](https://github.com/PeterSuh-Q3/tinycore-redpill/assets/5214574/b29902d7-4664-4866-9d27-69741a97ebaf)

Greetings,

Tried out recent release of 1.0.1.0 with Intel NUC6CAYH (J3455).
Encountered the following issue:
As photo illustrated, 
upon successful installation of tcrp xpenology, during first boot,
before initializing DSM, right after 'no more display' countdown concluded,
this line appears, "rops_handle_POLLIN_netlink: DELADDR"

Corresponding to this piece of log, the network connection terminated, link status LEDs are turned off.

Waited for half an hour, it seems the network LEDs came back, however, network never recovers. 
The NUC box can not ping through.

More interestingly, I have built another NUC6CAYH, the same model, two weeks ago, using 1.0.0.5 with flying colors.
I just simply repeated my steps with 1.0.1.0, and every tempts end up with "rops_handle_POLLIN_netlink: DELADDR" and network lights out.

Thanks in advance.

**Reply by PeterSuh-Q3** [reply]
The "rops_handle_POLLIN_netlink: DELADDR" message is not directly related to IP not being resolved.

I can't judge because I can't see the capture of the MAC address spoofing (switching between real MAC address and virtual MAC address) process above, but if you use spoofing to change the MAC address, the process of spoofing it again within DSM fixes the IP address within DSM. It plays a role.
However, I recently noticed that the attempted MAC spoofing addon within DSM was problematic.
https://github.com/PeterSuh-Q3/tcrp-addons/tree/main/mac-spoof
Also, since this problem was confirmed, this addon will no longer be used, so MAC spoofing and IP address fixation within DSM are not supported.

So, like RR, I plan to no longer support MAC spoofing in TCRP-MSHELL.
Instead, I plan to provide a separate function to change the NIC's real MAC address directly to users who want to change it to a Synology genuine MAC address.
If it is difficult to change it directly within MSHELL, you may need to use a utility, which is a solution that is already available.

As you can see, if MAC address spoofing is only handled in the FRIEND kernel and not in DSM, it will only lead to confusion for users like you.
Since there is no spoofing within DSM, it is highly likely that an incorrect IP associated with the real MAC address is assigned to the actual DSM.

Sometimes, even if you don't do MAC spoofing, the IP address is not fixed even within the real MAC address.

I think I should guide you depending on which of these two cases you fall into.

First, please try using the IP address that changes through finds.synology.com.

**Comment by wirwolf** [comment]
Hello. I download vmdk image and i can not boot interface
![tinycore-redpill-2024-01-07-12-12-21](https://github.com/PeterSuh-Q3/tinycore-redpill/assets/8457572/02aa4b04-62af-4222-af5e-5048df6c2936)
![tinycore-redpill-2024-01-07-12-12-35](https://github.com/PeterSuh-Q3/tinycore-redpill/assets/8457572/31c5f8e1-911e-4727-9383-01e74a7e94d2)

**Reply by wirwolf** [reply]
Also, I flash tinycore-redpill.v1.0.1.0.m-shell.img to a USB drive and use it in VMware as SCSI disk but i also see this screen

**Reply by PeterSuh-Q3** [reply]
How much memory did you allocate to TC in your virtual environment? 
It can also be a problem if the allocated memory is too low.

The above phenomenon occurs when the mydata.tgz file is not read properly at the "Loading extensions..." stage.

![스크린샷 2024-02-06 오후 12 06 19](https://github.com/PeterSuh-Q3/tinycore-redpill/assets/85427533/9a1d7ee4-6615-4360-ae54-e820f131ce60)

**Comment by gadkarisid** [comment]
Just wanted to say thanks for this project! I tried other tcrp VMDKs to build an Xpenology instance with no success, even after hours of troubleshooting. I was FINALLY able to get a working instance up with minimal effort using your project, so again, thanks!

**Reply by PeterSuh-Q3** [reply]
I'm glad it helped.
have a nice day.

**Comment by storm-99** [comment]
![image](https://github.com/PeterSuh-Q3/tinycore-redpill/assets/46198822/3c7c405c-6743-4fed-a28d-e10bb9531432)
Using VMWare Fusion, configured TCRP fine - TCRP loader as IDE:0, data disk as SCSI:0
but failed to install DSM, tried DSM 7.1 and 7.2, same error (7.2 failed at 5x%, 7.1 failed at 4x%), any idea?

**Reply by PeterSuh-Q3** [reply]
redpill does not yet support IDE and SCSI storage devices.
Use only SATA (AHCI) or NVMe.

![스크린샷 2024-03-18 오후 6 08 40](https://github.com/PeterSuh-Q3/tinycore-redpill/assets/85427533/cb51b3e6-c9a5-4659-9d77-281c7242fcc4)

**Reply by storm-99** [reply]
SATA works!!!
BTW, how about HBA (pass through) cards likse LSI 2308 under ESXi 7.03?
Configed similarly, with SataPortMap=18 DiskIdxMap=0C00 (don't want to show the SATA TCRP loader), but DSM install failed at 6x%
<img width="1582" alt="image" src="https://github.com/PeterSuh-Q3/tinycore-redpill/assets/46198822/babcf470-85b7-4bca-8c86-00239378a651">

**Reply by storm-99** [reply]
All sorted, got HBA card working. Need to convert the TCRP loader into flat vmdk in order to get the SATA working under ESXi.
One more thing, for DX3622xs, how to get the Storage Panel Size (Tower_8-bay) working? 
When building the loader, I already chose "Storage Panel Size (Tower_8-bay)", but Overview->Drive information still shows 12 bays.
And thank you for the great work!!!!

**Reply by PeterSuh-Q3** [reply, marked as answer]
Try editing and running the scheduler as described at the end of this notice.

https://xpenology.com/forum/topic/61839-tinycore-redpill-loader-build-support-tool-m-shell/?do=findComment&comment=449023

**Comment by storm-99** [comment]
Hi Peter,
It's me again. Anyone have problem with iSCSI? I can create iSCSI Lun and iSCSI Target fine, but just CANNOT map the iSCSI Target. It says
"LUN [veeamLUN-1] is not available due to load fail."
![image](https://github.com/PeterSuh-Q3/tinycore-redpill/assets/46198822/26cdc757-f90a-48ff-8a67-19cb0999a5bc)
If I reboot, the iSCSI Target shows offline, BUT I can map the iSCSI Target. 
![image](https://github.com/PeterSuh-Q3/tinycore-redpill/assets/46198822/928742cd-07b6-4dcb-a216-b42f5a4cf59a)
Tried Thick and Think Provisioning, same outcome.

**Reply by storm-99** [reply]
I am using DS3622xs+, DSM 7.2.1-69057 Update 4, loader v1.0.2.4
